providers/front-logger.js

```diff
@@ -0,0 +1,15 @@
+const { ipcRenderer } = require("electron");
+
+function logToString(log) {
```

Comment:
If we know it's either a log or a string, we can simplify it into `typeof log === "string" ? log : JSON.stringify(log, null, "\t")`